src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/KeyInfoX509Data.cs

```diff
@@ -155,7 +155,16 @@ public ArrayList IssuerSerials
 
         public void AddIssuerSerial(string issuerName, string serialNumber)
         {
-            BigInteger h = BigInteger.Parse(serialNumber, NumberStyles.AllowHexSpecifier);
+            if (string.IsNullOrEmpty(issuerName))
+                throw new ArgumentException(nameof(issuerName));
+
+            if (string.IsNullOrEmpty(serialNumber))
+                throw new ArgumentException(nameof(serialNumber));
+
+            BigInteger h;
+            if (!BigInteger.TryParse(serialNumber, NumberStyles.AllowHexSpecifier, NumberFormatInfo.CurrentInfo, out h))
+                throw new ArgumentException(nameof(serialNumber));
```

Comment:
new ArgumentException(string) is the message, not the variable name.  These should gain messages, or change to derived types (ArgumentNullException / ArgumentOutOfRangeException).  (The test can use ThrowsAny instead of Throws to catch it as ArgumentException)